lib/modules/rest/index.js

```diff
@@ -378,63 +377,25 @@ Rest.prototype.__buildCallMenu = function(exchangeDescription, endpoint, menu) {
   return callMenu;
 };
 
-Rest.prototype.__wrapExchange = util.promisify(function($happn, callback) {
-  var _this = this;
-
-  var cleanDescription = function(desc, exclusions) {
-    var cleaned = {};
+Rest.prototype.__updateDescription = function(description) {
+  const exclusions = ['initializing', 'setOptions', '_meta'];
 
-    exclusions.push('initializing');
-    exclusions.push('setOptions');
-    exclusions.push('_meta');
+  const cleanedDescription = Object.keys(description).reduce((cleanedDesc, key) => {
+    if (exclusions.indexOf(key) === -1);
```

Comment:
this does nothing, probably a stray semicolon making the if() useless

Reply:
ahh - unit test would have picked that up, which is what I am busy with...